hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/OutboxImpl.java

```diff
@@ -84,6 +89,10 @@ public final boolean offer(int ordinal, @Nonnull Object item) {
 
     @Override
     public final boolean offer(int[] ordinals, @Nonnull Object item) {
+        if (numRemainingInBatch <= 0) {
```

Comment:
Is there any legal way for `numRemainingInBatch` to become negative? If not, the condition should be `== 0` and it can be followed with an `assert numRemainingInBatch > 0`.

Reply:
It can be if the user ignores the return value and pushes items. Agree there should be some assert here.

Reply:
It doesn't seem possible since it's returned immediately if it's at 0. But agree with the assertion.